internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -422,6 +426,49 @@ func unpackRuleData(raw interface{}) ([]*gapi.AlertQuery, error) {
 	return result, nil
 }
 
+// normalizeModelJSON is the StateFunc for the `model`. It removes well-known default
+// values from the model json, so that users do not see perma-diffs when not specifying
+// the values explicitly in their Terraform.
+func normalizeModelJSON(model interface{}) string {
```

Comment:
We could consider adding an automated test for this.

The tests are all integration tests with actual .tf files, and they all implicitly verify that `plan` is empty when the tests are done. So, you could do this quickly by changing an existing test to have these fields missing.

The .tf files used in tests are in this dir: https://github.com/grafana/terraform-provider-grafana/tree/master/examples/resources/grafana_rule_group

Reply:
Julien's suggestion of a totally new test using extremely minimal JSON might be better long term :smile: